src/ai/susi/server/api/aaa/PasswordChangeService.java

```diff
@@ -62,55 +62,62 @@ public ServiceResponse serviceImpl(Query post, HttpServletResponse response, Aut
         String password = post.get("password", null);
         String newpassword = post.get("newpassword",null);
 
-
-        ClientCredential pwcredential = new ClientCredential(ClientCredential.Type.passwd_login, useremail);
-        Authentication authentication = DAO.getAuthentication(pwcredential);
-        ClientCredential emailcred = new ClientCredential(ClientCredential.Type.passwd_login,
-                authentication.getIdentity().getName());
-        ClientIdentity identity = authentication.getIdentity();
-        String passwordHash;
-        String salt;
-
-        try {
-            passwordHash = authentication.getString("passwordHash");
-            salt = authentication.getString("salt");
-        } catch (Throwable e) {
-            Log.getLog().info("Invalid password try for user: " + identity.getName() + " from host: " + post.getClientHost() + " : password or salt missing in database");
-            result.put("message", "invalid credentials");
-            throw new APIException(422, "Invalid credentials");
+        if(password.equals(newpassword)){
+            result.put("message", "Your current password and new password matches");
```

Comment:
Agreed, also as this message will be required in all clients so this is the way we should handle it.